lib/topological_inventory/providers/common/heartbeat.rb

```diff
@@ -0,0 +1,106 @@
+module TopologicalInventory
+  module Providers
+    module Common
+      class << self
+        attr_writer :heartbeat
+      end
+
+      def self.heartbeat(name)
+        @heartbeat ||= {}
+        @heartbeat[name] ||= TopologicalInventory::Providers::Common::Heartbeat.new(name)
+      end
+
+      module HeartbeatQueue
+        def heartbeat(name)
+          TopologicalInventory::Providers::Common.heartbeat(name)
+        end
+      end
+
+      class Heartbeat
+        HEARTBEAT_TIMEOUT = 10.freeze
```

Comment:
Should be increased to avoid false negative checks during kafka notifications and longer tower connection requests

https://github.com/RedHatInsights/topological_inventory-ansible_tower/blob/8e3b4a6e777b5a24e8e28ad5a0aa2a63900534b9/lib/topological_inventory/ansible_tower/connection.rb#L8

Reply:
this is covered by queuing mechanism on places where we are waiting on ansible tower client request (see 3 or 4 in description)

(it is 60s as default from Net::HTTP library - it is called read_timeout,
because in tower ruby client we can se only open_timeout (So Faraday's open_timeout is equivalent to Net::HTTP's open_timeout which is documented as:
>  Net::HTTP's open_timeout :Number of seconds to wait for the connection to open.
> And Faraday's timeout is equivalent to Net::HTTP's read_timeout which is documented as:
> 
>  Net::HTTP's read_timeout: Number of seconds to wait for one block to be read (via one read(2) call).

https://stackoverflow.com/questions/10322283/what-is-timeout-and-open-timeout-in-faraday )